Alignment/CommonAlignmentProducer/src/AlignmentProducerBase.cc

```diff
@@ -49,7 +47,29 @@ AlignmentProducerBase::AlignmentProducerBase(const edm::ParameterSet& config)
       saveDeformationsToDB_{config.getParameter<bool>("saveDeformationsToDB")},
       useSurvey_{config.getParameter<bool>("useSurvey")},
       enableAlignableUpdates_{config.getParameter<bool>("enableAlignableUpdates")},
-      idealGeometryLabel("idealForAlignmentProducerBase") {
+      idealGeometryLabel("idealForAlignmentProducerBase"),
+      ttopoToken_(iC.esConsumes<edm::Transition::BeginRun>()),
+      geomDetToken_(iC.esConsumes<edm::Transition::BeginRun>()),
+      ptpToken_(iC.esConsumes<edm::Transition::BeginRun>()),
+      //dtGeomToken_(iC.esConsumes<edm::Transition::BeginRun>(edm::ESInputTag("", "idealForAlignmentProducerBase"))),
+      //cscGeomToken_(iC.esConsumes<edm::Transition::BeginRun>(edm::ESInputTag("", "idealForAlignmentProducerBase"))),
+      //gemGeomToken_(iC.esConsumes<edm::Transition::BeginRun>(edm::ESInputTag("", "idealForAlignmentProducerBase"))),
```

Comment:
@cvuosalo @slomeo 
do you have any hunch as to why uncommenting these lines results in this error:
```
----- Begin Fatal Exception 03-Jun-2021 14:14:58 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Running EventSetup component MuonGeometryConstantsESModule/'muonGeometryConstants
Exception Message:
No data of type "DDCompactView" with label "" in record "IdealGeometryRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------
```
despite having the Muon geometry `ESProducers` included here:
https://github.com/cms-sw/cmssw/blob/a7f2449ccf521bd56aea903371f05064a0f23b6a/Alignment/CommonAlignmentProducer/python/AlignmentProducerAsAnalyzer_cff.py#L79-L103

?

Reply:
The `DDCompactViewESProducer` needs to be added:
```
process.DDCompactViewESProducer = cms.ESProducer("DDCompactViewESProducer",
                                                  appendToDataLabel = cms.string('')
)
```
Also, change all instances of `fromDDD` to False and `fromDD4hep` to True.

Reply:
@cvuosalo 
this results in:
```
----- Begin Fatal Exception 04-Jun-2021 09:25:50 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Running EventSetup component DDCompactViewESProducer/'DDCompactViewAlignmentProducerAsAnalyzer
Exception Message:
No data of type "DDDetector" with label "" in record "IdealGeometryRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------

```

Reply:
It is probably best to import `Configuration/​Geometry/​python/​GeometryDD4hep_cff.py` to get the necessary producers.

Reply:
@cvuosalo 
adding:
```diff
+from Configuration.Geometry.GeometryDD4hep_cff import *
```
results in:

```
---- Begin Fatal Exception 07-Jun-2021 13:04:47 CEST-----------------------
An exception of category 'ConfigFileReadError' occurred while
   [0] Processing the python configuration file named the.py
Exception Message:
 unknown python problem occurred.
RuntimeError: An exception of category 'FileInPathError' occurred.
Exception Message:
Relative path must not be empty


At:
  /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-06-03-1100/python/FWCore/ParameterSet/Types.py(835): insertInto
  /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-06-03-1100/python/FWCore/ParameterSet/Mixins.py(378): insertContentsInto
  /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-06-03-1100/python/FWCore/ParameterSet/Mixins.py(513): insertInto
  /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-06-03-1100/python/FWCore/ParameterSet/Config.py(1104): _insertManyInto
  /cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-06-03-1100/python/FWCore/ParameterSet/Config.py(1337): fillProcessDesc
  <string>(2): <module>

----- end Fatal Exception -------------------------------------------------
```
and if I add:

```
+DDDetectorESProducer.confGeomXMLFiles = cms.FileInPath("Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml")
```

I go back to initial error:
```
----- Begin Fatal Exception 07-Jun-2021 17:22:39 CEST-----------------------
An exception of category 'NoProxyException' occurred while
   [0] Running EventSetup component MuonGeometryConstantsESModule/'muonGeometryConstants
Exception Message:
No data of type "DDCompactView" with label "" in record "IdealGeometryRecord"
 Please add an ESSource or ESProducer to your job which can deliver this data.
----- End Fatal Exception -------------------------------------------------
```

Reply:
You might try adding:
```
 process.DDDetectorESProducer = cms.ESSource("DDDetectorESProducer",
                                             confGeomXMLFiles = cms.FileInPath('Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2021.xml'),
                                             appendToDataLabel = cms.string('')
                                             )
process.DDCompactViewESProducer = cms.ESProducer("DDCompactViewESProducer",
                                                  appendToDataLabel = cms.string('')
                                        )
```

Reply:
unfortunately none of the suggestions seems to work. As I don't have time to investigate further and the Tracker part is working as expected, I am opening the PR for review and I'll let he muon experts to tackle that.